src/meta/processors/admin/AdminClient.cpp

```diff
@@ -420,48 +420,67 @@ StatusOr<std::vector<HostAddr>> AdminClient::getPeers(GraphSpaceID spaceId, Part
     return Status::Error("Get Failed");
 }
 
+void AdminClient::getLeaderDist(HostAddr host,
+                                folly::Promise<StatusOr<storage::cpp2::GetLeaderResp>> pro,
+                                int32_t retry,
+                                int32_t retryLimit) {
+    auto* evb = ioThreadPool_->getEventBase();
+    folly::via(evb, [evb, host, pro = std::move(pro), retry, retryLimit, this] () mutable {
+        storage::cpp2::GetLeaderReq req;
+        auto client = clientsMan_->client(host, evb);
+        client->future_getLeaderPart(std::move(req))
+            .then(evb, [pro = std::move(pro), host, retry, retryLimit, this]
+                       (folly::Try<storage::cpp2::GetLeaderResp>&& t) mutable {
+            if (t.hasException()) {
+                LOG(ERROR) << folly::stringPrintf("RPC failure in AdminClient: %s",
+                                                  t.exception().what().c_str());
+                if (retry < retryLimit) {
+                    usleep(1000 * 50);
```

Comment:
maybe add a timer task to delay call getLeaderDist is better than sleep directly.

Reply:
Yeah, but it is not on main data path, we will modify all of them in AdminClient.